### PR TITLE
Allow full table paths to be used in joinable!

### LINKED ISSUE #333: Allow full path for tables in joinable! macro
It's currently only possible to pass an identifier as table to the joinable! macro (and also the #[belongs_to] and #[has_many] annotations).
Possible:

``` rust
use some_module::my_table_a;

joinable!(my_table_a -> my_table_b (some_field));
```

Not possible:

``` rust
joinable!(some_module::my_table_a -> my_table_b (some_field));
```

Results in:

```
error: no rules expected the token `::`
  --> src/lib.rs:19:28
19 |> joinable!(some_module::my_table_a -> my_table_b (some_field));
   |>                      ^^
```

### PR DESCRIPTION
**Pull request by Eijebong**
Fixes #333 

I'm not sure where to add tests as `joinable!` is not tested.

### CODE REVIEW

**Comment by sgrif** [comment]
It's probably worth adding some simple doctests to demonstrate its usage

**Comment by Eijebong** [comment]
Ok, so I added a doctest, not sure if it's enough

File: diesel/src/macros/mod.rs [hunk 1]
```diff
@@ -499,12 +499,43 @@ macro_rules! __diesel_table_query_source_impl {
...
+///
+/// # fn main() {
+/// use schema::*;
+///
+/// // Without the joinable! call, this wouldn't compile
+/// let query = users::table.inner_join(posts::table);
+/// # }
+/// ```
 #[macro_export]
 #[doc(hidden)]
```
Comment (killercup) [inline diesel/src/macros/mod.rs]: Eh, nice docs and all, but… ↑

Comment (Eijebong) [inline diesel/src/macros/mod.rs]: lol, @sgrif should I make this public ?

Comment (sgrif) [inline diesel/src/macros/mod.rs]: Yes

Comment (Eijebong) [inline diesel/src/macros/mod.rs]: Done

**Comment by killercup** [comment]
I'll merge this. I was gonna suggest mentioning it in the change log, but #948 will already do that.

**Pull request merged**